opensrp-reveal/src/main/java/org/smartregister/reveal/presenter/TaskRegisterFragmentPresenter.java

```diff
@@ -244,6 +258,68 @@ public void onIndexCaseFound(JSONObject indexCase, boolean isLinkedToJurisdictio
         }
     }
 
+    @Override
+    public void searchTasks(String searchText) {
+        Timber.d("searching task matching %s", searchText);
+        List<TaskDetails> filteredTasks = new ArrayList<>();
+        int withBuffer = 0;
+        for (TaskDetails task : tasks) {
+            if (Utils.matchesSearchPhrase(task.getFamilyName(), searchText) ||
+                    Utils.matchesSearchPhrase(task.getStructureName(), searchText) ||
+                    Utils.matchesSearchPhrase(task.getHouseNumber(), searchText) ||
+                    Utils.matchesSearchPhrase(task.getFamilyMemberNames(), searchText)) {
+                filteredTasks.add(task);
+                if (task.getDistanceFromUser() > 0 && task.getDistanceFromUser() <= Utils.getLocationBuffer())
+                    withBuffer++;
+            }
+        }
+        getView().setTaskDetails(filteredTasks);
+        getView().setTotalTasks(withBuffer);
+    }
+
+    @Override
+    public void filterTasks(TaskFilterParams filterParams) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method 'filterTasks(TaskFilterParams)' has an NPath complexity of 219](https://app.codacy.com/manual/OpenSRP/opensrp-client-reveal/pullRequest?prid=4684568)